src/main/java/com/google/sps/servlets/PostAnswerServlet.java

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.classes.SqlConstants;
+import com.google.sps.classes.Utility;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * This servlet will post an answer to a question.
+ * TODO(shaargtz): join this servlet with fetchAnswers into a single answer servlet.
+ */
+@WebServlet("/post-answer")
+public class PostAnswerServlet extends HttpServlet {
+
+  /** 
+   * Executes the query to insert an answer to the database.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String body = request.getParameter("answer-body");
+    int question_id = Integer.parseInt(request.getParameter("question_id"));
```

Comment:
Is it ever possible that this fails, or you get a bad question id?

Reply:
the preparedStatement takes care of parsing the string and escaping special characters, so no errors could happen on the query I believe. as for the id, if someone tries to go into /question.html?id=12312312 or something else that doesn't exist, it doesn't display anything, not even the form, and the form takes the ID directly from the URL string

Reply:
What I'm asking more is what happens if request.getParameter("question_id") isn't an integer? According to the [documentation](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#parseInt(java.lang.String)), you'll get a NumberFormatException. How are you handling that?

Reply:
there is no way that the parameter isn't an int, since the ID is taken directly from the URL string which has the ID parameter, and if the user tries going into a URL with a non-existing ID I just made it so that they get redirected to the index, so they cannot post the form; and for the actual exception, I'm using Utility.tryParseInt which is a try-catch and logs the exception instead of breaking the page

Reply:
btw the change to Utility.tryParseInt is in the next PR that is already ready to merge, that's why it doesn't show up here, but as soon as that is merged, the response will be empty instead of broken

Reply:
Got it, yeah in general you should practice defensive coding. Even though you have guaranteed that what this method tries to parse is an integer, what happens if someone makes a change down the road that breaks that guarantee? You should always catch exceptions and fail gracefully, and can note in a comment that it should never happen if that is indeed the case.

Reply:
okay, I merged the dependent PR, hope all of the other changes don't show up since they have already been approved